Testing/Unit tests.md

```diff
@@ -1,29 +1,99 @@
-A unit is the smallest logical piece of the application and unit tests are used to test it out. The tests should be as simple as possible and only cover the functionality of the unit. 
+The first type of automated tests most software developers encounter are unit tests. That's because the unit tests are an excellent tool for determining whether a specific code change or a new implementation caused some unexpected behaviors.
 
-Each test name should contain the name of the method it is testing, a particular condition and the expected result.
+The idea behind unit testing is to split the code into the smallest logical chunks - called _units_ - and test their functionalities separately. This means that the unit tests should be as simple as possible and cover only the functionality of that unit.
 
-```c#
-[Fact]
-public async Task Process_WhenParameterIsNotValid_ThenThrowArgumentException()
-{ }
+Their simplicity makes running all unit tests in a project quick and easy, which is why we should be checking them periodically during development time and especially before publishing our branch and creating a PR. We can go even further with that idea. By integrating test runs in the CI/CD pipelines and PR merge requirements we cannot forget to run them before sharing the code with others!
+
+Detailed unit tests help us validate that our code snippets do what they were meant to do, but they won't validate that we've implemented the requested functionalities successfully. To do that, all our units must work together, which is why we have other types of tests like [integration](integration-tests) and [UI](ui-tests) tests.
+
+### Tips and tricks
+
+#### Testing the current time
+
+Let's say that we have to write unit tests for the following method:
+
+``` c#
+public string GetTimeBasedGreeting() 
+{
+    var now = DateTime.UtcNow;
+
+    if (now.Hour >= 6 && now.Hour < 12)
+        return "Good morning";
+    
+    if (now.Hour >= 12 && now.Hour < 18)
+        return "Good afternoon";
```

Comment:
Since you are only using the `now` variable to fetch the `Hour` from it, maybe it makes more sense to declare something like `hourOfTheDay = DateTime.UtcNow.Hour` and then just use `hourOfTheDay` in the `if` statements, just a minor improvement for readability 😄

Reply:
Good point. I wanted to emphasize the usage of `now` in the code, but your suggestion will work as well while improving the readability.